folium/features.py

```diff
@@ -241,8 +241,19 @@ def __init__(self, data, style_function=None):
             else:  # This is a filename
                 self.embed = False
                 self.data = data
+        elif data.__class__.__name__ in ['GeoDataFrame','GeoSeries']:
+            if hasattr(data, '__geo_interface__'):
+                # We have a GeoPandas 0.2 object
+                self.embed = True
+                self.data = json.loads(json.dumps(data.to_crs(epsg='4326').__geo_interface__))
```

Comment:
> The `self.embed = True` could be up one level to avoid repetition

True. I do it.

>  `data.to_crs(epsg='4326')`  could be up one level to avoid repetition

No, otherwise, the Exception `'Unable to transform this object to a GeoJSON.'` will be replaced by `Unknown method 'to_crs'` (or something like this).

> PS: __geo_interface__ is a dict, right? Just making sure that we need the json.dumps.

`__geo_interface__` is a dict. But we need to do a deep copy of the object here ; otherwise, we'll have unintended side effects : further folium computation (like styling) may modify the ` geopandas`  object...